src/org/rascalmpl/test/infrastructure/TestFramework.java

```diff
@@ -28,23 +28,33 @@
 import java.util.Set;
 
 import org.junit.After;
+import org.rascalmpl.debug.IRascalMonitor;
+import org.rascalmpl.interpreter.ConsoleRascalMonitor;
 import org.rascalmpl.interpreter.Evaluator;
+import org.rascalmpl.interpreter.NullRascalMonitor;
 import org.rascalmpl.interpreter.env.GlobalEnvironment;
 import org.rascalmpl.interpreter.env.ModuleEnvironment;
 import org.rascalmpl.interpreter.load.StandardLibraryContributor;
 import org.rascalmpl.interpreter.result.Result;
 import org.rascalmpl.interpreter.staticErrors.StaticError;
+import org.rascalmpl.repl.TerminalProgressBarMonitor;
 import org.rascalmpl.uri.URIResolverRegistry;
 import org.rascalmpl.uri.URIUtil;
 import io.usethesource.vallang.IBool;
 import io.usethesource.vallang.ISourceLocation;
 import io.usethesource.vallang.IValue;
 import io.usethesource.vallang.exceptions.FactTypeUseException;
 import io.usethesource.vallang.type.TypeFactory;
+import jline.TerminalFactory;
+
 import org.rascalmpl.values.ValueFactoryFactory;
 
 
 public class TestFramework {
+	private final static IRascalMonitor monitor = System.console() != null 
```

Comment:
it's a bit of a pity we have to repeat this so many times, could we make a static function on the `TerminalProgressBarMonitor` class that does this?

Reply:
All I can think off is a specialized constructor for the Evaluator that takes this situation into account and then calls the other `this(...)` depending on the context. I'll give that a try.

Reply:
I'll give that a try. Don't like all this logic everywhere either.